language/move-prover/boogie-backend/src/boogie_helpers.rs

```diff
@@ -329,6 +329,48 @@ pub fn boogie_address_blob(_options: &BoogieOptions, val: &[BigUint]) -> String
     }
 }
 
+pub fn boogie_constant_blob(_options: &BoogieOptions, val: &[Constant]) -> String {
+    let args = val
+        .iter()
+        .map(|v| boogie_constant(_options, v))
+        .collect_vec();
+    if args.is_empty() {
+        "EmptyVec()".to_string()
+    } else {
+        boogie_make_vec_from_strings(&args)
+    }
+}
+
+pub fn boogie_constant(_options: &BoogieOptions, val: &Constant) -> String {
+    match val {
+        Constant::Bool(true) => "true".to_string(),
+        Constant::Bool(false) => "false".to_string(),
+        Constant::U8(num) => num.to_string(),
+        Constant::U64(num) => num.to_string(),
+        Constant::U128(num) => num.to_string(),
+        Constant::U256(num) => num.to_string(),
+        Constant::Address(v) => v.to_string(),
+        Constant::ByteArray(v) => boogie_byte_blob(_options, v),
+        Constant::AddressArray(v) => boogie_address_blob(_options, v),
+        Constant::Vector(vec) => boogie_make_vec_from_strings(
+            &vec.iter()
+                .map(|v| boogie_constant(_options, v))
+                .collect_vec(),
+        ),
+        Constant::U16(num) => num.to_string(),
+        Constant::U32(num) => num.to_string(),
+    }
+}
+
+pub fn boogie_value_blob(_options: &BoogieOptions, val: &[Value]) -> String {
+    let args = val.iter().map(|v| format!("{}", *v)).collect_vec();
```

Comment:
I think this `ast::Value` might also needs a helper function in boogie to replace the `format!("{}", *v)`